src/topology/builder.rs

```diff
@@ -157,6 +156,78 @@ pub async fn build_pieces(
     let (enrichment_tables, enrichment_errors) = load_enrichment_tables(config, diff).await;
     errors.extend(enrichment_errors);
 
+    let source_tasks = build_sources(
+        config,
+        diff,
+        &mut shutdown_coordinator,
+        &mut errors,
+        &mut outputs,
+        &mut tasks,
+    )
+    .await;
+
+    build_transforms(
+        config,
+        diff,
+        enrichment_tables,
+        &mut errors,
+        &mut inputs,
+        &mut outputs,
+        &mut tasks,
+    )
+    .await;
+
+    build_sinks(
+        config,
+        diff,
+        &mut errors,
+        buffers,
+        &mut inputs,
+        &mut healthchecks,
+        &mut tasks,
+        &mut detach_triggers,
+    )
+    .await;
```

Comment:
I'd like to see these repeated parameters moved into a common struct or something, and then the functions could be member functions of that struct. Notable, all of the following are repeated: `config`, `diff`, `errors`, `inputs`, `outputs`, and `tasks`.

Reply:
I really like this. It has made the diffs for the PR much harder to read, but it is largely just moving code around..